nodeup/pkg/model/protokube.go

```diff
@@ -58,15 +57,34 @@ func (t *ProtokubeBuilder) Build(c *fi.ModelBuilderContext) error {
 		return nil
 	}
 
-	if protokubeImage := t.NodeupConfig.ProtokubeImage[t.Architecture]; protokubeImage != nil {
-		c.AddTask(&nodetasks.LoadImageTask{
-			Name:    "protokube",
-			Sources: protokubeImage.Sources,
-			Hash:    protokubeImage.Hash,
-			Runtime: t.Cluster.Spec.ContainerRuntime,
-		})
+	// Move protokube bin to /usr/bin
+	protokubeBinName, protokubeBinRes, err := t.Assets.FindMatch(regexp.MustCompile(`protokube`))
+	if err != nil {
+		return err
 	}
 
+	fileTaskProtokubeBin := &nodetasks.File{
+		Path:     filepath.Join("/usr/bin", protokubeBinName),
+		Contents: protokubeBinRes,
+		Type:     nodetasks.FileType_File,
+		Mode:     fi.String("0755"),
+	}
+	c.AddTask(fileTaskProtokubeBin)
+
+	// Move channels bin to /. Protokube reads channels from /
+	channelBinName, channelBinRes, err := t.Assets.FindMatch(regexp.MustCompile(`channels`))
+	if err != nil {
+		return err
+	}
+
+	fileTaskChannel := &nodetasks.File{
+		Path:     filepath.Join("/", channelBinName),
+		Contents: channelBinRes,
+		Type:     nodetasks.FileType_File,
+		Mode:     fi.String("0755"),
+	}
+	c.AddTask(fileTaskChannel)
+
```

Comment:
Protokube must be modified to use the channels binary from "/opt/kops/bin" and the matches for the binaries would be nice to be more exact.
```suggestion
	// Copy protokube bin to /opt/kops/bin
	protokubeBinName, protokubeBinRes, err := t.Assets.FindMatch(regexp.MustCompile(`^protokube`))
	if err != nil {
		return err
	}
	fileTaskProtokubeBin := &nodetasks.File{
		Path:     filepath.Join("/opt/kops/bin", protokubeBinName),
		Contents: protokubeBinRes,
		Type:     nodetasks.FileType_File,
		Mode:     fi.String("0755"),
	}
	c.AddTask(fileTaskProtokubeBin)

	// Copy channels bin to /opt/kops/bin
	channelBinName, channelBinRes, err := t.Assets.FindMatch(regexp.MustCompile(`^channels$`))
	if err != nil {
		return err
	}
	fileTaskChannel := &nodetasks.File{
		Path:     filepath.Join("/opt/kops/bin", channelBinName),
		Contents: channelBinRes,
		Type:     nodetasks.FileType_File,
		Mode:     fi.String("0755"),
	}
	c.AddTask(fileTaskChannel)

```

Reply:
Sure will move the binaries to /opt/kops/bin